src/ucp/core/ucp_worker.c

```diff
@@ -385,6 +477,118 @@ void ucp_worker_progress(ucp_worker_h worker)
     ucs_assert(--worker->inprogress == 0);
 }
 
+ucs_status_t ucp_worker_get_efd(ucp_worker_h worker, int *fd)
+{
+    int res, res_fd;
+    ucs_status_t status;
+    ucp_rsc_index_t tl_id;
+    struct epoll_event event;
+    ucp_context_h context = worker->context;
+
+    if (worker->wakeup.wakeup_efd != -1) {
+        *fd = worker->wakeup.wakeup_efd;
+        return UCS_OK;
+    }
+
+    res_fd = epoll_create(context->num_tls);
+    if (res_fd == -1) {
+        ucs_error("Failed to create epoll descriptor");
+        return UCS_ERR_IO_ERROR;
+    }
+
+    event.events = EPOLLIN | EPOLLET;
```

Comment:
i would suggest to create separate function like "ucp_wakeup_context_add_fd" and use it here, and when adding uct fd